src/Application.php

```diff
@@ -136,6 +136,23 @@ public function runCommandList($commandList, $options)
             $exitCode = $command->run($this->outputFile);
             if ($exitCode) {
                 return $exitCode;
+            } else {
+                /**
+                 * Make sure the project directory is removed also since `composer remove` will leave behind
+                 * an empty composer.json file.  This can potentially cause the `cgr info` command to abort
+                 * prior to completing the process of iterating through all the projects.
+                 */
+                list(, $workingDir, $composerCommand, $orgProject) = explode(' ', $command->getCommandString());
+                if ($composerCommand == "'remove'") {
```

Comment:
1. Change `runCommandList()` so that it will run either function pointers or exec external tools.
2. Define a `removeCommand()` method, so that will be called instead of `generalCommand` for remove commands.
3. Make `removeCommand()` call through to `generalCommand()` and also return a function pointer to the extra cleanup you want to do.

Doing 1 implies a new base class for things to execute.